### PR TITLE
chore: add a story showing that disabling cols is possible

### PR DESCRIPTION
**Pull request by pierrebeitz**
we had a bug in disabling columns before and created an SSCCE before fixing it.
here's a more readable version of that.

### CODE REVIEW

**Comment by mperrotti** [comment]
@pierrebeitz - the columns are visually removed, but the `grid-template-columns` property stays the same. This means when I remove a column, all of the column widths remain the same.

**Comment by pierrebeitz** [comment]
@mperrotti uh, that's right! 💡🤔

do we want/need to handle that though? if developers show/hide stuff, we could encourage them to use `fr`s. if users show/hide columns they resized before, i think they will be fine with resizing again. i say that as i believe ui-kit should not try to automagically compute widths when changing column-visibility. but maybe you want to solve this differently (or have done this before - it just appears complex to me to implement that "correctly").

**Comment by pierrebeitz** [comment]
after some background process realized what you actually meant, i think i'll provide an update today.

**Comment by pierrebeitz** [comment]
should be fixed: 
table

File: packages/tablev2/Table.tsx [hunk 1]
```diff
@@ -231,30 +223,29 @@ export function Table<Entry, ColIDs extends string, Sort extends ColIDs>({
   const sort =
     col?.sorter?.(state.order, state.order === "asc" ? 1 : -1) ?? (() => 0);
   const sortedData = data.sort(sort);
+  const internalColumns: InternalColumn<Entry>[] = columns.map(c =>
+    Object.assign(c, { width: getWidth(c, state.columns) })
```
Comment (pierrebeitz) [inline packages/tablev2/Table.tsx:227]: this is the trick. we enhance the `Column` type by a width for internal use. that makes surrounding stuff less hard.

File: packages/tablev2/Table.tsx [hunk 2]
```diff
@@ -185,20 +185,17 @@ function HeaderRow<Entry>({
...
+  const className = cx(rowClassName(columns, stickyFirstCol), style.headerRow);
   return (
     <div role="row" className={className} key="headerRow">
       {columns.map(toHeaderCell)}
     </div>
   );
 }
 
+const getWidth = (col: Column<any>, state: State["columns"]) =>
+  state.find(s => s.id === col.id)?.width || col.initialWidth || "1fr";
```
Comment (pierrebeitz) [inline packages/tablev2/Table.tsx:197]: `getWidth` is now in charge of computing allTheWidths and their defaults.

**Comment by mperrotti** [comment]
So sorry for taking so long to come back to this! I missed your updates.
I'll do another review and try to get this merged today :)

**Comment by pierrebeitz** [comment]
corrected the failed linting... 🤦

**Comment by mperrotti** [comment]
@pierrebeitz - I think snapshots need to be updated

**Comment by GeorgiSTodorov** [comment]
@pierrebeitz , I updated the snapshots, I hope you don't mind.

**Comment by pierrebeitz** [comment]
@GeorgiSTodorov you know i'm always delighted when someone adds to my PRs! 💜  
Thanks a bunch and sorry for not updating those in the first place.

**Pull request merged**